### PR TITLE
BUG: Series.interpolate with dt64/td64 raises

### LINKED ISSUE #11312: ENH: interpolation of datetime values (NaT)
I didn't directly found an open issue about it, but is there a reason we do not implement interpolation of datetime _values_ (so not in the index)?

Example:

```
In [14]: s = pd.Series(pd.date_range('2012-01-01', periods=5))

In [16]: s[2] = np.nan

In [17]: s
Out[17]:
0   2012-01-01
1   2012-01-02
2          NaT
3   2012-01-04
4   2012-01-05
dtype: datetime64[ns]

In [18]: s.interpolate()
Out[18]:
0   2012-01-01
1   2012-01-02
2          NaT
3   2012-01-04
4   2012-01-05
dtype: datetime64[ns]
```

A crude manual work around:

```
In [20]: s2 = pd.Series(s.values.astype('int64'))

In [21]: s2[s2<0] = np.nan

In [22]: s2
Out[22]:
0    1.325376e+18
1    1.325462e+18
2             NaN
3    1.325635e+18
4    1.325722e+18
dtype: float64

In [23]: pd.to_datetime(s2.interpolate(), unit='ns')
Out[23]:
0   2012-01-01
1   2012-01-02
2   2012-01-03
3   2012-01-04
4   2012-01-05
dtype: datetime64[ns]
```

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [x] closes #11312 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/series/methods/test_interpolate.py [hunk 1]
```diff
@@ -78,6 +78,40 @@ def interp_methods_ind(request):
 
 
 class TestSeriesInterpolateData:
+    def test_interpolate_datetimelike_values(self):
```
Comment (mroeschke) [inline pandas/tests/series/methods/test_interpolate.py:81]: Do we have DataFrame.interpolate tests with NaT?

Comment (jbrockmendel) [reply pandas/tests/series/methods/test_interpolate.py:81]: i dont think so.  before long ill do a pass to parametrize all of these over frame_or_series

Comment (jbrockmendel) [reply pandas/tests/series/methods/test_interpolate.py]: good suggestion, turns out it was broken with DataFrame.  fixed.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**